apps/web/src/pages/workflow/views/editor/components/config-panel/components/param-input/index.tsx

```diff
@@ -1,8 +1,127 @@
+import { forwardRef, useImperativeHandle } from 'react';
+import { AddIcon, DeleteIcon } from '@milesight/shared/src/components';
+import { useI18n } from '@milesight/shared/src/hooks';
+import { IconButton, MenuItem, Select, TextField } from '@mui/material';
+import { Controller, useFieldArray, useForm } from 'react-hook-form';
+import { checkMaxLength } from '@milesight/shared/src/utils/validators';
+import './style.less';
+
+/**
+ * Form Data Types
+ */
+export type FormDataItem = {
+    arg: string;
+    type: number | undefined;
+};
+export type FormDataProps = {
+    params: FormDataItem[];
+};
 /**
  * Param Input Component
  *
  * Note: use in TriggerNode, CodeNode
  */
-const ParamInput = () => {};
+const Options = [
+    {
+        value: 0,
+        labelIntlKey: 'workflow.node.config_panel_type_string',
+    },
+    {
+        value: 1,
+        labelIntlKey: 'workflow.node.config_panel_type_long',
+    },
+    {
+        value: 2,
+        labelIntlKey: 'workflow.node.config_panel_type_double',
+    },
+    {
+        value: 3,
+        labelIntlKey: 'workflow.node.config_panel_type_boolean',
+    },
+    {
+        value: 4,
+        labelIntlKey: 'workflow.node.config_panel_type_binary',
+    },
+];
+type ParamInputProps = {
+    title?: string;
+};
+const dynamicFields = 'params';
+const ParamInput = forwardRef(({ title }: ParamInputProps, ref) => {
+    const { getIntlText } = useI18n();
+
+    // Initialize react-hook-form
+    // TODO: set defaultValue
+    const { control, handleSubmit, reset } = useForm<FormDataProps>();
+
+    // Manage dynamic fields
+    const { fields, append, remove } = useFieldArray({
+        control,
+        name: dynamicFields,
+    });
+
+    // Add new row
+    const addParamRow = () => {
+        append({
+            arg: '',
+            type: undefined,
+        });
+    };
+
+    // Expose methods to parent component
+    useImperativeHandle(ref, () => ({
+        handleSubmit,
+        resetForm: () => reset(),
+    }));
```

Comment:
The form component is not well designed and should expose the value, onChange properties to allow the parent to perform form data processing.